app/controllers/idv/in_person/address_controller.rb

```diff
@@ -5,14 +5,36 @@ class AddressController < ApplicationController
 
       before_action :render_404_if_in_person_residential_address_controller_enabled_not_set
       before_action :confirm_in_person_state_id_step_complete
-      before_action :confirm_in_person_address_step_needed
+      before_action :confirm_in_person_address_step_needed, only: :show
+      before_action :confirm_ssn_step_needed
 
       def show
         analytics.idv_in_person_proofing_address_visited(**analytics_arguments)
 
         render :show, locals: extra_view_variables
       end
 
+      def update
+        attrs = Idv::InPerson::AddressForm::ATTRIBUTES
+
+        attrs = attrs.difference([:same_address_as_id])
+        pii_from_user[:same_address_as_id] = 'false' if updating_address?
+
+        attrs.each do |attr|
+          pii_from_user[attr] = flow_params[attr]
+        end
+
+        form_result = form.submit(flow_params)
+
+        analytics.idv_in_person_proofing_residential_address_submitted(**form_result.to_h)
+
+        if updating_address?
```

Comment:
@soniaconnolly since I'm doing this check here and redirecting to verify_info do I still need the check mentioned in this [comment](https://github.com/18F/identity-idp/pull/9377#discussion_r1369364625)?

Reply:
You'll have to test it out to make sure, but I believe that you do, because the before_action will redirect away from the Ssn step if the user already has an address in the session, the way they do when they're editing from VerifyInfo. It looks like you did add it below.